FWCore/Framework/src/StreamSchedule.cc

```diff
@@ -378,6 +381,154 @@ namespace edm {
+                    auto branches = conditionalModuleBranches.equal_range(productModuleLabel);
+                    for (auto itBranch = branches.first; itBranch != branches.second; ++it) {
+                      if (it->second.originalInstanceLabel == "*" or
+                          itBranch->second->productInstanceName() == it->second.originalInstanceLabel) {
+                        if (typeIsViewCompatible(ci.type(),
+                                                 TypeID(itBranch->second->wrappedType().typeInfo()),
+                                                 itBranch->second->className())) {
+                          productFromConditionalModule = true;
+                          break;
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+            itFound = conditionalModules.find(productModuleLabel);
+          } else {
+            //need to check the rest of the data product info
+            auto findBranches = conditionalModuleBranches.equal_range(productModuleLabel);
+            for (auto itBranch = findBranches.first; itBranch != findBranches.second; ++itBranch) {
+              if (itBranch->second->productInstanceName() == ci.instance()) {
+                if (ci.kindOfType() == PRODUCT_TYPE) {
+                  if (ci.type() == itBranch->second->unwrappedTypeID()) {
+                    productFromConditionalModule = true;
+                    break;
+                  }
+                } else {
+                  //this is a view
+                  if (typeIsViewCompatible(ci.type(),
+                                           TypeID(itBranch->second->wrappedType().typeInfo()),
+                                           itBranch->second->className())) {
+                    productFromConditionalModule = true;
+                    break;
+                  }
+                }
+              }
+            }
+          }
+          if (productFromConditionalModule) {
+            auto condWorker =
+                getWorker(productModuleLabel, proc_pset, workerManager_, preg, prealloc, processConfiguration);
+            assert(condWorker);
+
+            conditionalModules.erase(itFound);
+
+            auto dependents = tryToPlaceConditionalModules(condWorker,
+                                                           conditionalModules,
+                                                           conditionalModuleBranches,
+                                                           aliasMap,
+                                                           proc_pset,
+                                                           preg,
+                                                           prealloc,
+                                                           processConfiguration);
+            returnValue.insert(returnValue.end(), dependents.begin(), dependents.end());
+            returnValue.push_back(condWorker);
```

Comment:
You missed the fact that as tryToPlaceConditionalModules it slowly removes conditional modules from the list. Therefore what you'd actually get is
```
a b f1 c f2
```

Reply:
Ah right, the insertion is being done here only if `productFromConditionalModule == true`. For non-alias branches the logic is clear. But for the alias branches, I think, the alias-to module could be inserted many times. (or did I miss something again?)

In the consumesMany branch I see elements being removed from `conditionalModules`, but there is no other action done with it. I believe that would lead to duplicate entries in the Path.

Reply:
Fixed for consumesMany